From dc6eb55b8d8d1ab3d53634e36cd8c62367522771 Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Date: Tue, 19 Mar 2019 15:49:08 +0200
Subject: [PATCH 478/741] staging: fsl-dpaa2/mac: Fix bug in pause frame
 advertising

commit 4f96e7ef823b5927f145cc7deabbc82a855ba4bf from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Fix issue where advertised pause frame settings may be overwritten.

Signed-off-by: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/staging/fsl-dpaa2/mac/mac.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/staging/fsl-dpaa2/mac/mac.c b/drivers/staging/fsl-dpaa2/mac/mac.c
index 577bfb731b0a..ec6935225d45 100644
--- a/drivers/staging/fsl-dpaa2/mac/mac.c
+++ b/drivers/staging/fsl-dpaa2/mac/mac.c
@@ -418,6 +418,11 @@ static void configure_link(struct dpaa2_mac_priv *priv,
 	phydev->speed = cfg->rate;
 	phydev->duplex  = !!(cfg->options & DPMAC_LINK_OPT_HALF_DUPLEX);
 
+	if (cfg->advertising != 0) {
+		linkmode_zero(phydev->advertising);
+		link_mode_dpmac2phydev(cfg->advertising, phydev->advertising);
+	}
+
 	if (linkmode_test_bit(ETHTOOL_LINK_MODE_Pause_BIT, phydev->supported)) {
 		if (cfg->options & DPMAC_LINK_OPT_PAUSE)
 			linkmode_set_bit(ETHTOOL_LINK_MODE_Pause_BIT, phydev->advertising);
@@ -432,11 +437,6 @@ static void configure_link(struct dpaa2_mac_priv *priv,
 			linkmode_clear_bit(ETHTOOL_LINK_MODE_Asym_Pause_BIT, phydev->advertising);
 	}
 
-	if (cfg->advertising != 0) {
-		linkmode_zero(phydev->advertising);
-		link_mode_dpmac2phydev(cfg->advertising, phydev->advertising);
-	}
-
 	if (cfg->options & DPMAC_LINK_OPT_AUTONEG) {
 		phydev->autoneg = AUTONEG_ENABLE;
 		linkmode_set_bit(ETHTOOL_LINK_MODE_Autoneg_BIT, phydev->advertising);
-- 
2.17.1

